src/devtools/useAtomsDevtools.ts

```diff
@@ -0,0 +1,212 @@
+    }
+
+    result[atomToPrintable(atom)] = [...atomState.d.keys()].map(atomToPrintable)
+  })
+
+  return result
+}
+
+export function useAtomsDevtools(name: string, scope?: Scope) {
+  const snapshot = useAtomsSnapshot()
+  const goToSnapshot = useGotoAtomsSnapshot()
+  let extension: Extension | undefined
+  try {
+    extension = (window as any).__REDUX_DEVTOOLS_EXTENSION__ as Extension
+  } catch {
+    // ignored
+  }
+  if (!extension) {
+    if (
+      typeof process === 'object' &&
+      process.env.NODE_ENV === 'development' &&
+      typeof window !== 'undefined'
+    ) {
+      console.warn('Please install/enable Redux devtools extension')
+    }
+  }
+
+  const ScopeContext = getScopeContext(scope)
+  const scopeContainer = useContext(ScopeContext)
+  const store = scopeContainer.s
+
+  if (!store[DEV_SUBSCRIBE_STATE]) {
+    throw new Error('useAtomsSnapshot can only be used in dev mode.')
+  }
+
+  const isTimeTraveling = useRef(false)
+  const isRecording = useRef(true)
+  const devtools = useRef<ConnectionResult & { shouldInit?: boolean }>()
+
+  const snapshots = useRef<AtomsSnapshot[]>([])
+
+  useEffect(() => {
+    if (extension) {
+      const connection = extension.connect({ name })
+      const devtoolsUnsubscribe = connection.subscribe((message: Message) => {
+        switch (message.type) {
+          case 'DISPATCH':
+            switch (message.payload?.type) {
+              case 'RESET':
+                // Todo
+                return
+              case 'COMMIT': {
+                const lastSnapshot = snapshots.current[
+                  snapshots.current.length - 1
+                ] as AtomsSnapshot
+
+                const serializedSnapshot = serializeSnapshot(lastSnapshot)
+
+                connection.init({
+                  values: serializedSnapshot,
+                  dependencies: getDependencies(store, lastSnapshot),
+                })
+                snapshots.current.length = 0
+                return
+              }
+
+              case 'JUMP_TO_ACTION': {
+                isTimeTraveling.current = true
+
+                const currentSnapshot = snapshots.current[
+                  message.payload.actionId - 1
+                ] as AtomsSnapshot
+
+                goToSnapshot(currentSnapshot)
+                return
+              }
+
+              case 'PAUSE_RECORDING':
+                isRecording.current = !isRecording.current
+                return
+            }
+        }
+      })
+
+      devtools.current = connection
+      devtools.current.shouldInit = true
+      return devtoolsUnsubscribe
+    }
+  }, [store, extension, goToSnapshot, name])
+
+  useEffect(() => {
+    if (!devtools.current) {
+      return
+    }
+    if (devtools.current.shouldInit) {
+      devtools.current.init(undefined)
+      devtools.current.shouldInit = false
+      return
+    }
+
+    if (!snapshot.size) {
+      return
+    }
+
+    if (isTimeTraveling.current) {
+      queueMicrotask(() => {
```

Comment:
we never use queueMicrotask anywhere else. while it's technically fine, i'm concerned and curious why we don't need it with `useAtomDevtools`.

Reply:
I tried different batching solutions, but only `queueMicrotask` worked since it's a little bit other than Promise.resolve().then & setTimeout